util/dvsim/SGE.py

```diff
@@ -0,0 +1,349 @@
+                if len(t) >= 5 and t[0] == str(jobid):
+                    # Find a line with useful info
+                    if re.search(t[4], 'qwrt'):
+                        # Job must be queued, running or being transferred
+                        dowait = True
+                        break
+                    if re.search(t[4], 'acuE'):
+                        # Job or host in error state
+                        logger.warning('Job %d in error state', str(jobid))
+
+            if dowait:
+                time.sleep(interval)
+                if name is None:
+                    logger.info("Time %s: waiting for jobid %s to finish",
+                                time.ctime(), str(jobid))
+                else:
+                    logger.info("Time %s: waiting for job '%s' (jobid %s) to \
+finish", time.ctime(), name, str(jobid))
+
+    def submit(self, job, array=False, useenvironment=True, usecwd=True,
+               name=None, stdin=None, stdout=None, stderr=None,
+               joinstdouterr=True, nproc=1, wait=True, lammpi=True):
+        """
+        Submits a job to SGE
+        Returns jobid as a number
+        """
+
+        logger = logging.getLogger('SGE.submit')
+
+        logger.info("Submitting job:   " + str(job) + " stdout: %s \
+Stderr: %s", stdout, stderr)
+
+        # Parameters to qsub specified as the header of the job specified on
+        # STDIN
+        lamstring = lammpi and f" -pe lammpi {nproc}" or ""
+        qsuboptslist = ['-cwd -V ', lamstring]
+
+        if name is not None:
+            qsuboptslist.append('-N ' + name)
+        if stdin is not None:
+            qsuboptslist.append('-i ' + stdin)
+        if stdout is not None:
+            qsuboptslist.append('-o ' + stdout)
+        if stderr is not None:
+            qsuboptslist.append('-e ' + stderr)
+        if joinstdouterr:
+            qsuboptslist.append('-j')
+        if wait:
+            qsuboptslist.append('-sync y')
+        if usecwd:
+            qsuboptslist.append('-cwd')
+        if useenvironment:
+            qsuboptslist.append('-V')
+        if array is not False:
+            try:
+                n = int(array[0])
+            except IndexError:
+                n = int(array)
+                raise IndexError("List is empty!")
+            except ValueError:
+                error_msg = "Could not convert data to an integer."
+                logger.error(error_msg)
+                raise ValueError(error_msg)
+            try:
+                m = int(array[1])
+            except ValueError:
+                m = None
+                raise ValueError("Could not convert data to an integer.")
+            except IndexError:
+                m = None
+                raise IndexError("Could not convert data to an integer.")
+            try:
+                s = int(array[2])
+            except IndexError:
+                s = None
+                raise IndexError("Could not convert data to an integer.")
```

Comment:
An IndexError exception is due to array[2] being an out of bounds access, not a conversion error.

Reply:
changed the error message